packages/account/src/Components/icon-message-list/__tests__/icon-message-list.spec.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Icon } from '@deriv/components';
+import IconMessageList from '../icon-message-list';
+
+jest.mock('@deriv/components', () => {
+    const original_module = jest.requireActual('@deriv/components');
+    return {
+        ...original_module,
+        Icon: jest.fn(() => <div data-testid='mocked_icon'></div>),
+    };
+});
+describe('<IconMessageList/>', () => {
+    const messages_list = ['Sample Text1', 'Sample Text2', 'Sample Text3', 'Sample Text4'];
+    it('should render IconMessageList component', () => {
+        render(<IconMessageList />);
+        expect(screen.getByTestId('icon_message_list')).toBeInTheDocument();
+    });
+
+    it('should render icon passed to the component', () => {
+        render(<IconMessageList icon={<Icon icon='sampleIcon' />} />);
+        expect(screen.getByTestId('mocked_icon')).toBeInTheDocument();
+    });
+    it('should show message passed to the component', () => {
+        render(<IconMessageList message={'Lorem Ipsom'} />);
+        expect(screen.getByText(/lorem ipsom/i)).toBeInTheDocument();
+    });
+    it('when the length of message_list is less than 3, it should show messages with icon ', () => {
+        render(<IconMessageList message_list={['Sample Text1', 'Sample Text2']} />);
+        expect(screen.getByText(/Sample Text1/i)).toBeInTheDocument();
+        expect(screen.getByText(/Sample Text2/i)).toBeInTheDocument();
+    });
+
+    it('should show first 3 msgs and show_more_btn when the message_list is more than 3', () => {
+        render(<IconMessageList message_list={messages_list} />);
+        expect(screen.getByText(/sample text1/i)).toBeInTheDocument();
+        expect(screen.getByText(/sample text2/i)).toBeInTheDocument();
+        expect(screen.getByText(/sample text3/i)).toBeInTheDocument();
+        expect(screen.queryByText(/sample text4/i)).not.toBeInTheDocument();
+        expect(
+            screen.getByRole('button', {
+                name: /show more/i,
+            })
+        ).toBeInTheDocument();
```

Comment:
What about checking 'show less' button is not present in the document?

Reply:
Done.